lib/puppet/resource/status.rb

```diff
@@ -144,7 +144,7 @@ def failed_because(detail)
       # Puppet::Transaction::Event failure with the given message.
       # @param message [String] the reason for a status failure
       def fail_with_event(message)
-        failed = true
+        self.failed = true
```

Comment:
I need to investigate this more, as it appears the failed state for the event isn't set correctly.

Reply:
This was introduced in https://github.com/puppetlabs/puppet/commit/eea93dd872856a7a3bf70d21f87473f091071a12

Reply:
The line is extraneous because we correctly set `self.failed` when adding the `failure` event in https://github.com/puppetlabs/puppet/blob/eea93dd872856a7a3bf70d21f87473f091071a12/lib/puppet/resource/status.rb#L46. So line 147 can be safely deleted.